rules/linux/execution_shell_via_udp_cli_utility_linux.toml

```diff
@@ -0,0 +1,89 @@
+UDP reverse shell to bypass traditional firewall restrictions and gain remote access to a target system covertly.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-auditd_manager.auditd-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Reverse Shell via UDP"
+note = """## Setup
+This rule requires the use of the `auditd_manager` integration. `Auditd_manager` is a tool designed to simplify and enhance the management of the audit subsystem in Linux systems. It provides a user-friendly interface and automation capabilities for configuring and monitoring system auditing through the auditd daemon. With `auditd_manager`, administrators can easily define audit rules, track system events, and generate comprehensive audit reports, improving overall security and compliance in the system. The following steps should be executed in order to install and deploy `auditd_manager` on a Linux system. 
+```
+Kibana -->
+Management -->
+Integrations -->
+Auditd Manager -->
+Add Auditd Manager
+```
+`Auditd_manager` subscribes to the kernel and receives events as they occur without any additional configuration. However, if more advanced configuration is required to detect specific behavior, audit rules can be added to the integration in either the "audit rules" configuration box or the "auditd rule files" box by specifying a file to read the audit rules from. 
+```
+For this detection rule no additional audit rules are required to be added to the integration. 
+```
+Add the newly installed `auditd manager` to an agent policy, and deploy the agent on a Linux system from which auditd log files are desirable.
+"""
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 47
+rule_id = "a5eb21b7-13cc-4b94-9fe2-29bb2914e037"
+severity = "medium"
+tags = ["OS: Linux", "Use Case: Threat Detection", "Tactic: Execution"]
+type = "eql"
+query = '''
+sample by host.id, process.pid, process.parent.pid
+[ process where host.os.type == "linux" and auditd.data.syscall == "execve" and 
+  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "perl", "python*", 
+  "nc", "ncat", "netcat", "php*", "ruby", "openssl", "awk", "telnet", "lua*", "socat") ]
+[ process where host.os.type == "linux" and auditd.data.syscall == "socket" and 
+  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "perl", "python*",
+  "nc", "ncat", "netcat", "php*", "ruby", "openssl", "awk", "telnet", "lua*", "socat") and 
+  auditd.data.a0 == 2 and auditd.data.a1 : ("2", "802") ]
+[ network where host.os.type == "linux" and auditd.data.syscall == "connect" and 
+  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "perl", "python*",
+  "nc", "ncat", "netcat", "php*", "ruby", "openssl", "awk", "telnet", "lua*", "socat") and 
+  network.direction == "egress" ]
```

Comment:
```suggestion
[ process where host.os.type == "linux" and event.dataset == "auditd_manager.auditd" and auditd.data.syscall == "execve" and 
  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "perl", "python*", 
  "nc", "ncat", "netcat", "php*", "ruby", "openssl", "awk", "telnet", "lua*", "socat") ]
[ process where host.os.type == "linux" and event.dataset == "auditd_manager.auditd" and auditd.data.syscall == "socket" and 
  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "perl", "python*",
  "nc", "ncat", "netcat", "php*", "ruby", "openssl", "awk", "telnet", "lua*", "socat") and 
  auditd.data.a0 == 2 and auditd.data.a1 : ("2", "802") ]
[ network where host.os.type == "linux" and event.dataset == "auditd_manager.auditd" and auditd.data.syscall == "connect" and 
  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "perl", "python*",
  "nc", "ncat", "netcat", "php*", "ruby", "openssl", "awk", "telnet", "lua*", "socat") and 
  network.direction == "egress" ]
```

Whenever we use an integration, we have to set the `event.dataset` field and value in the query. This allows the rule validation to parse the package `auditd_manager` and the integration data stream `auditd`. It then uses these to check if those fields exist by loading the correct schema and matching on keys.

This was causing the original error where the field `auditd.data.syscall` was not found. After adding this, you can see that it knows which package is being used now.

<img width="933" alt="Screenshot 2023-07-05 at 9 42 47 AM" src="https://github.com/elastic/detection-rules/assets/99630311/6ab9c4dd-0c5b-4c5a-ae2f-f981f2bd98f4">